x/programs/cmd/simulator/src/lib.rs

```diff
@@ -155,51 +169,10 @@ impl From<Key> for Param {
     }
 }
 
-#[derive(Debug, Serialize, PartialEq)]
-pub struct Plan<'a> {
-    /// The key of the caller used in each step of the plan.
-    pub caller_key: &'a str,
-    /// The steps to perform in the plan.
-    pub steps: Vec<Step>,
-}
-
-impl<'a> Plan<'a> {
-    /// Pass in the `caller_key` to be used in each step of the plan.
-    #[must_use]
-    pub fn new(caller_key: &'a str) -> Self {
-        Self {
-            caller_key,
-            steps: vec![],
-        }
-    }
-
-    /// returns the [Id] of the added [Step]
-    pub fn add_step(&mut self, step: Step) -> Id {
-        self.steps.push(step);
-        Id::from(self.steps.len() - 1)
-    }
-}
-
-#[derive(Debug, Deserialize)]
-pub struct BaseResponse {
-    /// The numeric id of the step.
-    pub id: u32,
-    /// An optional error message.
-    pub error: Option<PlanError>,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct PlanError(String);
-impl core::fmt::Debug for PlanError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0)
-    }
-}
-
 #[derive(Debug, Deserialize)]
-pub struct PlanResult {
+pub struct StepResult {
```

Comment:
Can you rename `Step` to `Action`? A "step" only makes sense in the context of a plan (a plan has many steps). A simulator doesn't really have steps; it executes actions.

Reply:
I'm scared that this is going to need a lot of changes, it's also called a step on the go side. Should these be equivalent to actions triggered by transactions ? If not, then we may need to find another name to avoid any confusion

Reply:
I think that we need to discuss more about renaming plans